### ISSUE TITLE
[Bug]: performance.measureUserAgentSpecificMemory causes excetion

### ISSUE THREAD

**Issue by masa-nuc** [electron/electron#37147]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

22.2.0

### What operating system are you using?

Windows

### Operating System Version

Windows 10 Pro, Version 22H2, OS build 19045.2546

### What arch are you using?

x64

### Expected Behavior

It should perform the performance.measureUserAgentSpecificMemory function without error, if crossOriginIsolated is true.

### Actual Behavior

It causes exception:
DOMException: Failed to execute 'measureUserAgentSpecificMemory' on 'Performance': performance.measureUserAgentSpecificMemory is not available.

### Testcase Gist URL

https://github.com/masa-nuc/ua-specific-memory-test

**Comment by masa-nuc** [comment]
I have tested [this](https://github.com/masa-nuc/ua-specific-memory-test) with latest Electron v23.1.1 and it also causes same error.

Hi @VerteDinde , is there any update? thanks.

**Comment by 0xdevalias** [comment]
I get:

```js
performance.measureUserAgentSpecificMemory()

// Uncaught TypeError: performance.measureUserAgentSpecificMemory is not a function
```

On electron version `Electron/24.1.2`:

```js
navigator.userAgent

// 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Beeper/3.52.29 Chrome/112.0.5615.87 Electron/24.1.2 Safari/537.36'
```

**Comment by 0xdevalias** [comment]
Some more details/context from other issues:

> @nornagon I think you were sort of missing the point of this issue when you closed it.
> 
> ---
> 
> Relevant context from the other issue where this originated from:
> 
> > NB https://github.com/WICG/performance-measure-memory seems relevant. Enable with `enableBlinkFeatures: 'MeasureMemory'`. Does that solve your problem?
> > 
> > _Originally posted by @nornagon in https://github.com/electron/electron/issues/30383#issuecomment-892194040_
> 
> > @nornagon I think I need to set some CORS policies or something to get that working, I'm not quite sure how to do that yet, I'll update you once I can figure that out.
> > 
> > _Originally posted by @fabiospampinato in https://github.com/electron/electron/issues/30383#issuecomment-893775990_
> 
> > @nornagon Sorry for the ping, I can't find any resource on how to make the renderer process cross origin isolated, which my understanding is is a requirement for being able to use this new API. Do you know what the solution here is?
> > 
> > _Originally posted by @fabiospampinato in https://github.com/electron/electron/issues/30383#issuecomment-1267079565_
> 
> > > how to make the renderer process cross origin isolated
> > 
> > renderer processes are isolated per origin by default. how are you managing to make them _not_ cross origin isolated?
> > 
> > _Originally posted by @nornagon in https://github.com/electron/electron/issues/30383#issuecomment-1267236122_
> 
> > @nornagon I've opened an issue about this [here](https://github.com/electron/electron/issues/35905). I'm getting `globalThis.crossOriginIsolated === false` in a plain fiddle using Electron v21 🤷‍♂️
> > 
> > _Originally posted by @fabiospampinato in https://github.com/electron/electron/issues/30383#issuecomment-1267249805_
> 
> ---
> 
> **tl;dr**:
> 
> - @nornagon suggested using `performance.measureUserAgentSpecificMemory`, but that is only available when `crossOriginIsolated` is `true` ([Ref](https://github.com/electron/electron/pull/30383#issuecomment-892194040))
> - @nornagon suggested that "renderer processes are isolated per origin by default", implying that `crossOriginIsolated` should be `true`, yet that isn't the case, as @fabiospampinato stated, which resulted in this issue being opened (Ref: [1](https://github.com/electron/electron/pull/30383#issuecomment-1267236122), [2](https://github.com/electron/electron/pull/30383#issuecomment-1267249805))
> 
> This also seems to be the same root cause as the following issue which has since been opened, and remains an issue in at least electron `24.1.2` as far as I can tell:
> 
> - https://github.com/electron/electron/issues/37147
> 
> _Originally posted by @0xdevalias in https://github.com/electron/electron/issues/35905#issuecomment-1532292862_

**Comment by 0xdevalias** [comment]
Not sure if/how well it will apply to electron, but just found this:

- https://peter.sh/experiments/chromium-command-line-switches/
  - `--enable-precise-memory-info`: Make the values returned to window.performance.memory more granular and more up to date in shared worker. Without this flag, the memory information is still available, but it is bucketized and updated less frequently. This flag also applys to workers.

**Comment by jtbandes** [comment]
I'm running into this issue too. I'm using electron with a webpack dev server configured to send the relevant headers to enable cross-origin isolation:

<img width="611" alt="image" src="https://github.com/electron/electron/assets/14237/54aab5d8-1c94-4871-8ada-18f53fb838a0">

And indeed `crossOriginIsolated` is true, yet this API still does not work:

<img width="1045" alt="image" src="https://github.com/electron/electron/assets/14237/f092441a-a36b-4844-a6ad-64a29439ad21">


I've tried adding `enableBlinkFeatures: "MeasureMemory",` to `webPreferences` and that didn't work. Has anyone found a workaround?

**Comment by jtbandes** [comment]
This is still an issue as far as I know!

**Comment by jtbandes** [comment]
I believe this is still an issue.

**Comment by bouk** [comment]
I looked into it a bit deeper, I think this might be resolved by setting up a performance manager in `ElectronBrowserMainParts::PostCreateThreads`, like in the `shell_browser` in chromium:

https://source.chromium.org/chromium/chromium/src/+/main:content/shell/browser/shell_browser_main_parts.cc;l=190;drc=c0265133106c7647e90f9aaa4377d28190b1a6a9;bpv=1;bpt=1

https://github.com/electron/electron/blob/7ec813732aff4bc428e4946efb1934cc9f6a1fd6/shell/browser/electron_browser_main_parts.cc#L366

Then the `performance_manager.mojom.DocumentCoordinationUnit` gets initialized I believe

**Comment by bouk** [comment]
I've created a PR at https://github.com/electron/electron/pull/42267

**Comment by Mekacher-Anis** [comment]
still having this issue, excited to see it working :D will the PR be merged?

**Comment by Mekacher-Anis** [comment]
@bouk what's still missing for the PR to get merged?
